cmd/skaffold/app/cmd/run.go

```diff
@@ -47,7 +47,18 @@ func doRun(ctx context.Context, out io.Writer) error {
 			return fmt.Errorf("failed to build: %w", err)
 		}
 
+		if !opts.SkipTests {
+			err = r.Test(ctx, out, bRes)
+			if err != nil {
+				return fmt.Errorf("failed to test: %w", err)
+			}
+		}
+
 		err = r.DeployAndLog(ctx, out, bRes)
+		if err != nil {
+			return fmt.Errorf("failed to deploy: %w", err)
+		}
+
 		if err == nil {
 			tips.PrintForRun(out, opts)
 		}
```

Comment:
nit: `return nil` instead of `return err`

Reply:
Updated.